packages/graphql-dynamodb-transformer/src/definitions.ts

```diff
@@ -144,16 +156,14 @@ export function makeCreateInputObject(
     })
     .map((field: FieldDefinitionNode) => {
       let type: TypeNode;
-      if (field.name.value === 'id') {
+      const fieldName = field.name.value;
+      if (
+        Object.keys(autoGeneratableFieldsWithType).indexOf(fieldName) !== -1 &&
+        autoGeneratableFieldsWithType[fieldName].indexOf(unwrapNonNull(field.type).name.value) !== -1
+      ) {
         // ids are always optional. when provided the value is used.
         // when not provided the value is not used.
-        type = {
-          kind: Kind.NAMED_TYPE,
-          name: {
-            kind: Kind.NAME,
-            value: 'ID',
-          },
-        };
+        type = unwrapNonNull(field.type);
```

Comment:
is this change related to this PR?

Reply:
Yes, this code snippet makes the field to be nullable if they are autogeneratable